src/Uno.UI.RemoteControl.Host/RemoteControlServer.cs

```diff
@@ -44,7 +44,13 @@ private AssemblyLoadContext GetAssemblyLoadContext(string applicationId)
 		{
 			if (_loadContexts.TryGetValue(applicationId, out var lc))
 			{
-				_loadContexts.TryUpdate(applicationId, (lc.Context, lc.Count + 1), (lc.Context, lc.Count));
+				if (!_loadContexts.TryUpdate(applicationId, (lc.Context, lc.Count + 1), (lc.Context, lc.Count)))
+				{
+					if (this.Log().IsEnabled(LogLevel.Debug))
+					{
+						this.Log().LogDebug($"Failed to increment the count of load context connections for '{applicationId}'");
+					}
+				}
```

Comment:
This may cause memory leaks in case of concurrency issues. 

To fix this, we either need to either:
- move to optimistic retries to redo the `TryGetValue` to get an updated value until the `TryUpdate` works (for this location and the one below)
- use a synchronization root with a simple `private object _loadContextGate = new()` and move to a `Dictionary<,>` instead of a `ConcurrentDictionary<,>` then use `lock()` statements where appropriate.

Either implementation can work, but I'd lean over to the lock for now to make the logic simpler (as long as there is only  
`_loadContext` to synchronize)